dask/tests/test_highgraph.py

```diff
@@ -57,3 +59,51 @@ def test_cull():
 
     culled_by_y = hg.cull({"y"})
     assert dict(culled_by_y) == a
+
+
+@pytest.mark.parametrize("inject_dict", [True, False])
+def test_map_basic_layers(inject_dict):
+    """Check map_basic_layers() by injecting an inc() call"""
+
+    y = da.ones(3, chunks=(3,), dtype="int") + 40
+
+    def inject_inc(dsk):
+        assert isinstance(dsk, BasicLayer)
+        dsk = dict(dsk)
+        k = next(iter(dsk))
+        dsk[k] = (inc, dsk[k])
+        if inject_dict:
+            return dsk  # map_basic_layers() should automatically convert it to a `BasicLayer`
+        else:
+            return BasicLayer(dsk)
+
+    dsk = y.__dask_graph__()
+    y.dask = dsk.map_basic_layers(inject_inc)
+    layers = list(y.dask.layers.values())
+    assert type(layers[0]) == BasicLayer
+    assert type(layers[1]) == Blockwise
```

Comment:
Nitpick: I _think_ this should be equivalent for what we're trying to test

```suggestion
    assert isinstance(layers[0], BasicLayer)
    assert isinstance(layers[1], Blockwise)
```